### ISSUE TITLE
 Unexpected error: [Errno 13] Permission denied: '.public_suffix_list.dat'

### ISSUE THREAD

**Issue by TheWinz** [domainaware/parsedmarc#54]
I am on Ubuntu 18.04 . parsedmarc version 6.0.1
the imap part of the parsedmarc service is retrieving reports from inbox fine. It cannot pass to elasticsearch.
I get an unexpected error because parsedmarc apparently cannot get to the .public_suffix_list.dat file. 
I have reproduced this on 2 Ubuntu systems. I also get this error when I try to pass the test aggregate report. Example of the error:

Feb 09 19:24:04 kvw-dmarc parsedmarc[12718]:  WARNING:__init__.py:1037:Message with subject "Aggregate test" is not a valid aggregate DMARC report: Unexpected error: [Errno 13] Permission denied: '.public_suffix_list.dat'

I can pass the same report manually. When I do that I get:

root@kvw-dmarc:~# parsedmarc -c parsedmarctest.ini file_path acme.org\!example.com\!1335571200\!1335657599.zip
   DEBUG:elastic.py:281:Saving aggregate report to Elasticsearch
   DEBUG:elastic.py:208:Creating Elasticsearch index: dmarc_aggregate-2012-04-28

The report passes.
When I do an ls -la I see 2 files added in the directory where I called parsedmarc:
.GeoLite2-Country.mmdb
.public_suffix_list.dat

I am wondering if this has something to do with the change to /etc/parsedmarc.ini and the parsedmarc user/group. I understand this change from a security perspective and applaud it. I am curious though if this is a permission issue and that is why I propose this.

Your help in solving this mystery is much appreciated.

Greeting from the Netherlands!

**Comment by seanthegeek** [comment]
Hi,

This indicates that the user you are running the service as does not have permission to create the `.public_suffix_list.dat` file in the service's current working directory (CWD).

Check the CWD of your service.

I'll look into using a temp directory instead of the CWD.

**Comment by seanthegeek** [comment]
This fix will be included in `parsedmarc` 6.0.2 which will be released as soon as I fix issue #53.

If you would like this fix now, upgrade to the latest development Git commit, by following the instructions in the documentation.

**Comment by seanthegeek** [comment]
@TheWinz  You mentioned here that you are still getting this error in 6.0.2 unless you set the cwd in the service configuration file. https://github.com/domainaware/parsedmarc/issues/55#issuecomment-462656511

That is very strange, because starting in 6.0.2, `parsedmarc` does not write to the CWD at all, and instead creates its own directory under `/tmp`, which it deletes at exit.

I've noticed that sometimes `pip` does not notice the new release right away.

Also, if you are using a virtualenv (as recommended in the instructions for using pypy3),make sure you are upgrading inside the virtualenv. For example:

```bash
sudo -H /opt/venvs/parsedmarc/bin/pip3 install -U parsedmarc
```
Then try restarting the service without the cwd option.
